### PR TITLE
Add Http2 support & envoy

### PR DESCRIPTION
**Pull request by mkocher**
We've updated the release to be able to test an http2 gorouter and updated gostatic to have an envoy in the data path so that it mimics the real world.

We're currently running four benchmarks in the errand - against gorouter h1, gorouter h2, envoy and static. We may want to change the assertions at the end to assert that both h1 and h2 tests are fast enough.

todo: 
- [x] Upload the envoy blob somewhere
- [ ] decide how much of this should run in CI
- [ ] consider deleting the jupyter notebook stuff, it's broken maybe

### CODE REVIEW

File: jobs/performance_tests/templates/run_perf_test.erb
Comment (ctlong) [inline jobs/performance_tests/templates/run_perf_test.erb:70]: Do we really want to delete this function? And if we do, then we should probably remove the call to it on line 104 of this same file.

Comment (ameowlia) [reply jobs/performance_tests/templates/run_perf_test.erb:70]: We don't use datadog anymore that I know of. This is safe to delete.

File: jobs/performance_tests/templates/run_perf_test.erb
Comment (ameowlia) [inline jobs/performance_tests/templates/run_perf_test.erb:43]: It looks like the performance tests currently always run h2 tests, but according to this [PR](https://github.com/cloudfoundry/routing-release/pull/205) gorouter will not always have h2 enabled. And currently, since the PR is not merged, no gorouter supports h2 and his test will fail.

❓ Can you make it configurable to run the h2 tests?

File: jobs/performance_tests/templates/run_perf_test.erb
Comment (ameowlia) [inline jobs/performance_tests/templates/run_perf_test.erb:54]: ❓ Can you explain what this is testing? What is a "gostatic envoy" (this line) vs a "gostatic direct load test" (below)

File: manifests/bran.yml
Comment (ameowlia) [inline manifests/bran.yml:2]: ❓ what is bran?

**Review by ameowlia — CHANGES_REQUESTED** [review]
I'm so glad you all are adding envoy into the performance test path.

❓ Can you make sure to edit the readme before this is merged in. Feel free to completely trash the current one, since most of it seems no longer true or relevant.

File: manifests/bran.yml
Comment (ctlong) [reply manifests/bran.yml:2]: @ameowlia it's Mona's favorite kind of muffin, and the arbitrary name we chose for our manifest. We could probably stand to change this.

File: jobs/performance_tests/templates/run_perf_test.erb
Comment (ctlong) [reply jobs/performance_tests/templates/run_perf_test.erb:54]: "gostatic envoy" hits the app through envoy using HTTP/1.1 and TLS. "gostatic direct load test" hits the app directly via HTTP/1.1, circumventing envoy.

The purpose IIRC was to determine how much envoy TLS termination was slowing down our requests.

File: jobs/performance_tests/templates/run_perf_test.erb
Comment (ctlong) [reply jobs/performance_tests/templates/run_perf_test.erb:43]: It might be that we don't look to merge this until after Gorouter does support h2. Are there cases (or CI) where performance tests are run on older versions of Gorouter?

Comment (ameowlia) [reply jobs/performance_tests/templates/run_perf_test.erb:43]: We do not run performance tests against old versions of gorouter.

However, the current PR to routing release and gorouter makes it seem like we are going to have a bosh property to enable h2. If we have a switch there to turn it off, I think we should have a switch here to turn it off too.

File: jobs/performance_tests/templates/run_perf_test.erb
Comment (ameowlia) [reply jobs/performance_tests/templates/run_perf_test.erb:54]: Oh, nice. That makes a lot of sense. Can you add something to that effect in the comments so it is more clear?

File: manifests/bran.yml
Comment (ameowlia) [reply manifests/bran.yml:2]: 😹  yeah, please change. Also the correct favorite type of muffin is banana nut muffin 🍌 .

File: jobs/performance_tests/templates/run_perf_test.erb
Comment (mkocher) [reply jobs/performance_tests/templates/run_perf_test.erb:70]: cool, we'll delete *all* the datadog stuff from the release when we clean it up

File: manifests/bran.yml
Comment (mkocher) [reply manifests/bran.yml:2]: Gotta go with @monamohebbi here, Bran muffins are way better than banana nut muffins.

As part of the cleanup phase we want to delete all the other manifests and just leave one modern bosh 2.0 manifest.

Comment (ctlong) [reply manifests/bran.yml:2]: I would argue that a muffin without any chocolate in it cannot be the best muffin...

File: jobs/performance_tests/templates/run_perf_test.erb
Comment (ameowlia) [reply jobs/performance_tests/templates/run_perf_test.erb:70]: 👍

**Comment by ameowlia** [comment]
Do you plan on taking this out of "draft" @mkocher, @ctlong ?

**Comment by ctlong** [comment]
@ameowlia we don't have a plan to do so at the moment – I don't remember whether this branch is ready to commit, but I think it probably isn't.

**Comment by geofffranks** [comment]
@ctlong @mkocher seems like there were still some minor changes needed for this, right? Is this PR we want to follow up on + get it merged?

**Comment by ctlong** [comment]
@geofffranks this work stalled after the routing-perf-tests stopped being actively run in CI. IIRC we thought that more general performance testing guidelines were being talked about and wanted to wait for those.

My inclination is to slim down the PR to some choice changes (ie. deleting datadog, adding envoy) and call it quits there. I don't trust the `hey` fork we made to work with HTTP/2.

TBH I'm even fine just closing the PR and leaving the branch for posterity in the off chance we start using this repo again.

**Pull request closed without merging**
